src/SquidMath.h

```diff
@@ -123,5 +123,40 @@ SetToNaturalSumOrMax(S &var, Args... args)
     return var;
 }
 
+template <typename P, typename T, typename U>
```

Comment:
We will need to document this function, including the odd fact that the first parameter _value_ is unused.